xmr-btc/tests/on_chain.rs

```diff
@@ -0,0 +1,67 @@
+mod harness;
+
+use anyhow::Result;
+use genawaiter::GeneratorState;
+use harness::wallet::{bitcoin, monero};
+use xmr_btc::{
+    action_generator_bob,
+    bitcoin::{BroadcastSignedTransaction, SignTxLock},
+    bob,
+    monero::CreateWalletForOutput,
+    Action, ReceiveTransferProof,
+};
+
+struct Network;
+
+impl ReceiveTransferProof for Network {
+    fn receive_transfer_proof(&self) -> xmr_btc::monero::TransferProof {
```

Comment:
This will need to be `async`.

You can probably implement this by having a oneshot where you return the receiver here, keep the sender around and when you get the event from libp2p, pass it in into the sender to continue the protocol.

Reply:
You may want this to be `&mut self` for that.